lemmy_api/src/user.rs

```diff
@@ -1305,3 +1308,59 @@ impl Perform for UserJoin {
     Ok(UserJoinResponse { joined: true })
   }
 }
+
+#[async_trait::async_trait(?Send)]
+impl Perform for GetReportCount {
+  type Response = GetReportCountResponse;
+
+  async fn perform(
+    &self,
+    context: &Data<LemmyContext>,
+    websocket_id: Option<ConnectionId>,
+  ) -> Result<GetReportCountResponse, LemmyError> {
+    let data: &GetReportCount = &self;
+    let user = get_user_from_jwt(&data.auth, context.pool()).await?;
+
+    let user_id = user.id;
+    let community_id = data.community;
+    let community_ids =
+      collect_moderated_communities(user_id, community_id, context.pool()).await?;
```

Comment:
I just realized we're doing this really inefficiently... the smarter way to do it (rather than doing 2 queries, first one to get modded communities), would be to join to the `community_moderator` table, and have all the report counts and queries do a `for_mod_id` thing.... But don't worry about this now, we wanna get rid of the views anyway.